packages/core/EventManager.ts

```diff
@@ -439,7 +439,7 @@ export default class EventManager {
         });
       }
 
-      if (newBooking) {
+      if (newBooking?.references.length) {
         calendarReference = newBooking.references.find((reference) => reference.type.includes("_calendar"));
       } else {
         // Bookings should only have one calendar reference
```

Comment:
```suggestion
       calendarReference = newBooking?.references.length
         ? newBooking.references.find((reference) => reference.type.includes("_calendar"))
         : booking.references.find((reference) => reference.type.includes("_calendar"));
```
Minor NIT suggestion